basic_bootloader/src/bootloader/mod.rs

```diff
@@ -237,6 +240,12 @@ impl<S: EthereumLikeTypes> BasicBootloader<S> {
 
             tracer.begin_tx(initial_calldata_buffer);
 
+            // Take a snapshot in case we need to invalidate the
+            // transaction to seal the block.
+            // This can happen if any of the block limits (native, gas, pubdata
+            // logs) is reached by the current transaction.
+            let rollback_handle = system.start_global_frame()?;
```

Comment:
Maybe we should use some more descriptive name? So it will be easier to understand it below

Reply:
Makes sense, changed it to pre_tx_rollback_handle